solutions/usaco-739.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: usaco-739
+source: USACO Silver 2017 US Open 
+title: Bovine Genomics
+author: Brad Ma
+---
+
+[Official Editorial (C++)](http://www.usaco.org/current/data/sol_cownomics_silver_open17.html)
+
+## Solution
+
+**Time Complexity** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+This implementation uses sets, which has not been covered yet.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class BovineGenomics {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+		//BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		//PrintWriter pw = new PrintWriter(System.out);
+
+		int n = io.nextInt();
+		int m = io.nextInt();
+
+		int[][] spotted = new int[n][m];
+		for (int x = 0; x < n; x++) {  // read spotted
+			StringBuilder line = new StringBuilder();
```

Comment:
Why are you using a stringbuilder if you're only appending to it once?

Reply:
good practice to use string builders on string problems, .charAt can be expensive

Reply:
.charAt has never failed me?

Reply:
yeah I don't think the charAt method in strings causes Java to TLE, normally it's i/o

Reply:
.charAt is O(1) for Strings as well. [See this](https://stackoverflow.com/a/6461439)

Reply:
good practice to use string builders and there are constants